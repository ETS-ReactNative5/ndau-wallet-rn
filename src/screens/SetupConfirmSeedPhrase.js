import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  ScrollView,
  Text,
  Button,
  ProgressViewIOS,
  Platform,
  ProgressBarAndroid,
  SafeAreaView,
  TextInput,
  TouchableHighlight
} from 'react-native';

var _ = require('lodash');

class SetupConfirmSeedPhrase extends Component {
  constructor(props) {
    super(props);
    this.state = {
      seedPhraseFromSelection: '',
      textColor: '#ffffff',
      showErrorText: false,
      errorWord: '',
      errorCount: 0,
      selectedPhrase: '',
      match: false,
      firstThree: [],
      secondThree: [],
      thirdThree: [],
      fourthThree: []
    };
  }

  componentDidMount = () => {
    this.props.navigator.setStyle({
      drawUnderTabBar: true,
      tabBarHidden: true
    });
    //add default props
    for (item in this.props.seedPhraseArray) {
      this.setState({ [`${item}BackgroundColor`]: '#1c2227' });
    }

    //shuffle the deck
    let shuffledArray = this.shuffleArray(this.props.seedPhraseArray);
    this.setState({
      firstThree: shuffledArray.slice(0, 3),
      secondThree: shuffledArray.slice(3, 6),
      thirdThree: shuffledArray.slice(6, 9),
      fourthThree: shuffledArray.slice(9, 12)
    });
  };

  onPushAnother = () => {
    this.props.navigator.push({
      label: 'SetupTermsOfService',
      screen: 'ndau.SetupTermsOfService',
      passProps: {
        encryptionPassword: this.props.encryptionPassword,
        userId: this.props.userId,
        parentStyles: this.props.parentStyles,
        iconsMap: this.props.iconsMap,
        numberOfAccounts: this.props.numberOfAccounts,
        seedPhraseArray: this.props.seedPhraseArray
      },
      navigatorStyle: {
        drawUnderTabBar: true,
        tabBarHidden: true
      }
    });
  };

  render() {
    return (
      <SafeAreaView style={styles.safeContainer}>
        <View style={styles.container}>
          <ScrollView style={styles.contentContainer}>
            <View>
              <Text style={this.props.parentStyles.wizardText}>Confirm seed phrase</Text>
            </View>
            <View>
              {Platform.OS === 'android' ? (
                <ProgressBarAndroid
                  styleAttr="Horizontal"
                  progress={0.625}
                  style={this.props.parentStyles.progress}
                  indeterminate={false}
                />
              ) : (
                <ProgressViewIOS progress={0.625} style={this.props.parentStyles.progress} />
              )}
            </View>
            <View>
              <Text style={this.props.parentStyles.wizardText}>
                Demonstrate that you wrote the phrase down by tapping the words below in order.{' '}
              </Text>
            </View>
            <TextInput
              style={{
                height: 70,
                borderColor: 'gray',
                borderWidth: 1,
                marginBottom: 10,
                marginTop: 10,
                paddingLeft: 10,
                color: '#000000',
                backgroundColor: '#ffffff',
                fontSize: 18,
                fontFamily: 'TitilliumWeb-Regular'
              }}
              value={this.state.seedPhraseFromSelection}
              placeholder=""
              placeholderTextColor="#333"
              multiline={true}
              numberOfLines={2}
              editable={false}
            />
            {this.state.showErrorText ? this.state.errorCount < 4 ? (
              <View style={styles.errorContainer}>
                <Text style={styles.errorText}>
                  Please enter the words in the correct order. De-select the last word to continue.{' '}
                </Text>
              </View>
            ) : (
              <View style={styles.errorContainer}>
                <Text style={styles.errorText}>
                  Please click the Back button to generate a new seed phrase. Write down your phrase
                  instead of memorizing it, or you may lose access to your ndau.{' '}
                </Text>
              </View>
            ) : null}
            <View style={styles.rowView}>
              {this.state.firstThree.map((item, index) => {
                return this.showWord(index, item);
              })}
            </View>
            <View style={styles.rowView}>
              {this.state.secondThree.map((item, index) => {
                return this.showWord(index, item);
              })}
            </View>
            <View style={styles.rowView}>
              {this.state.thirdThree.map((item, index) => {
                return this.showWord(index, item);
              })}
            </View>
            <View style={styles.rowView}>
              {this.state.fourthThree.map((item, index) => {
                return this.showWord(index, item);
              })}
            </View>
          </ScrollView>
          <View style={styles.footer}>
            <Button
              color="#4d9678"
              onPress={this.onPushAnother}
              title="Next"
              disabled={!this.state.match}
            />
          </View>
        </View>
      </SafeAreaView>
    );
  }

  // shuffleArray implements a Fisher-Yates shuffle algorithm;
  // walks through the array backwards once, exchanging each value
  // with a random element from the remainder of the array
  shuffleArray(b) {
    let a = b.slice();
    for (let i = a.length - 1; i >= 0; i--) {
      let r = Math.floor(Math.random() * i);
      // now swap r and i
      [ a[i], a[r] ] = [ a[r], a[i] ];
    }
    return a;
  }

  confirmPhraseOrder() {
    const seedPhraseFromSelectionArray = this.state.seedPhraseFromSelection.split(' ');
    if (
      this.props.seedPhraseArray[seedPhraseFromSelectionArray.length - 2] !==
      this.state.selectedPhrase
    ) {
      this.setState({
        [`${this.state.selectedPhrase}BackgroundColor`]: '#ff0000',
        showErrorText: true,
        errorCount: this.state.errorCount + 1,
        errorWord: this.state.selectedPhrase
      });
    } else {
      this.setState({
        [`${this.state.selectedPhrase}BackgroundColor`]: '#0000ff',
        showErrorText: false,
        errorWord: ''
      });
    }
  }

  handleClick(item) {
    if (this.state.showErrorText) {
      if (item === this.state.errorWord && this.state.errorCount < 4) {
        let newTWPFromSelection = this.state.seedPhraseFromSelection.substring(
          0,
          this.state.seedPhraseFromSelection.lastIndexOf(
            ' ',
            this.state.seedPhraseFromSelection.length - 2
          )
        );
        this.setState(
          {
            seedPhraseFromSelection: newTWPFromSelection + ' '
          },
          () => {
            this.setState({
              [`${this.state.selectedPhrase}BackgroundColor`]: '#1c2227',
              showErrorText: false,
              errorWord: '',
              selectedPhrase: ''
            });
            this.confirmPhraseOrder;
          }
        );
      } else {
        return;
      }
    } else if (this.state.seedPhraseFromSelection.indexOf(item) !== -1) {
      return;
    } else {
      const newValue = this.state.seedPhraseFromSelection + item;
      const newValueArray = newValue.split(' ');
      newValue += ' ';
      const match = _.isEqual(this.props.seedPhraseArray, newValueArray);
      this.setState(
        {
          seedPhraseFromSelection: newValue,
          selectedPhrase: item,
          match: match
        },
        this.confirmPhraseOrder
      );
    }
  }

  showWord(index, item) {
    return (
      <TouchableHighlight key={index} onPress={(event) => this.handleClick(item, event)}>
        <View
          style={{
            height: 40,
            width: 100,
            marginBottom: 10,
            marginTop: 10,
            backgroundColor: eval(`this.state.${item}BackgroundColor`)
          }}
        >
          <Text
            style={{
              color: '#ffffff',
              fontSize: 20,
              textAlign: 'center'
            }}
          >
            {item}
          </Text>
        </View>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  safeContainer: {
    flex: 1,
    backgroundColor: '#1c2227'
  },
  container: {
    flex: 1,
    paddingLeft: 10,
    paddingTop: 10,
    paddingRight: 10,
    paddingBottom: 10,

    backgroundColor: '#1c2227'
  },

  contentContainer: {
    flex: 1 // pushes the footer to the end of the screen
  },
  footer: {
    justifyContent: 'flex-end'
  },
  progress: {
    paddingTop: 30,
    paddingBottom: 30
  },
  rowView: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-evenly'
  },
  errorText: {
    color: '#f75f4b',
    fontSize: 20
  },
  errorContainer: {
    backgroundColor: '#f5d8d1'
  }
});

export default SetupConfirmSeedPhrase;
