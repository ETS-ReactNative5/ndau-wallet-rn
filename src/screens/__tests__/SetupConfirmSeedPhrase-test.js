import React from 'react';
import { StyleSheet } from 'react-native';
import SetupConfirmSeedPhrase from '../SetupConfirmSeedPhrase';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

function makeStyles() {
  return StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  });
}

function makeNavigator() {
  return {
    setStyle: () => {},
    toggleNavBar: () => {}
  };
}

function makeWords() {
  return 'zero one two three four five six seven eight nine ten eleven'.split(' ');
}

describe('SetupConfirmSeedPhrase presentation', () => {
  beforeEach(() => {
    this.tree = renderer
      .create(
        <SetupConfirmSeedPhrase
          navigator={makeNavigator()}
          parentStyles={makeStyles()}
          seedPhraseArray={makeWords()}
          encryptionPassword={'testing'}
          entropy={'dGVzdGluZ3dlc3Rpbmdh'}
          shuffledWords={makeWords()}
          shuffleMap={[ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 ]}
        />
      )
      .toJSON();
  });

  test('renders correctly', () => {
    expect(this.tree).toMatchSnapshot();
  });
});

describe('SetupConfirmSeedPhrase behavior', () => {
  beforeEach(() => {
    // set up default wrapper for every test.
    this.wrapper = mount(
      <SetupConfirmSeedPhrase
        navigator={makeNavigator()}
        parentStyles={makeStyles()}
        seedPhraseArray={makeWords()}
        encryptionPassword={'testing'}
        entropy={'dGVzdGluZ3dlc3Rpbmdh'}
        shuffledWords={makeWords()}
        shuffleMap={[ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 ]}
      />
    );
    this.press = (index) => {
      this.wrapper.find('Word').at(index).props().onPress();
    };
  });

  test('done button starts disabled', () => {
    const doneButton = this.wrapper.find('Button').at(1);
    expect(doneButton.prop('disabled')).toBeTruthy();
  });

  test('trigger error state with incorrect sequence', () => {
    this.press(0);
    this.press(1);
    this.press(2);
    this.press(4); // wrong
    this.wrapper.update();
    expect(this.wrapper.state().inError).toBeTruthy();
  });

  test('error count goes up with incorrect input', () => {
    this.press(4); // wrong
    this.wrapper.update();
    expect(this.wrapper.state().errorWord).toBe(4);
    expect(this.wrapper.state().errorCount).toBe(1);
  });

  test('done button is enabled after correct sequence', () => {
    makeWords().forEach((e, i) => {
      this.wrapper.find('Word').at(i).props().onPress();
    });
    this.wrapper.update();
    const doneButton = this.wrapper.find('Button').at(1);
    expect(doneButton.prop('disabled')).toBeFalsy();
  });

  test('trigger error state with incorrect word', () => {
    this.press(6); // wrong
    this.wrapper.update();
    expect(this.wrapper.state().inError).toBeTruthy();
    expect(this.wrapper.state().errorWord).toBe(6);
    expect(this.wrapper.state().errorCount).toBe(1);
  });

  test('correct error state by deselecting word', () => {
    this.press(6); // wrong
    this.press(6); // correction
    this.wrapper.update();
    expect(this.wrapper.state().inError).toBeFalsy();
    expect(this.wrapper.state().errorCount).toBe(1);
  });

  it('should not be able to click any buttons after a mistake', () => {
    this.press(6); // wrong
    this.press(1); // another wrong
    this.wrapper.update();
    expect(this.wrapper.state().selected).toEqual([ 6 ]);
    expect(this.wrapper.state().errorCount).toBe(1);
  });

  test('trigger error state with incorrect sequence', () => {
    this.press(0);
    this.press(1);
    this.press(2);
    this.press(4); // wrong
    this.wrapper.update();
    expect(this.wrapper.state().inError).toBeTruthy();
    expect(this.wrapper.state().errorWord).toBe(4);
    expect(this.wrapper.state().errorCount).toBe(1);
  });

  test('lockem out after too many mistakes', () => {
    this.press(1); // wrong
    expect(this.wrapper.state().errorCount).toBe(1);
    this.press(1); // correction

    this.press(2); // wrong
    expect(this.wrapper.state().errorCount).toBe(2);
    this.press(2); // correction

    this.press(0); // correct

    this.press(4); // wrong
    expect(this.wrapper.state().errorCount).toBe(3);
    this.press(4); // correction

    this.press(4); // wrong again

    expect(this.wrapper.state().errorCount).toBe(4);
    this.wrapper.update();
    expect(this.wrapper.state().mustRetry).toBeTruthy();
  });
});
