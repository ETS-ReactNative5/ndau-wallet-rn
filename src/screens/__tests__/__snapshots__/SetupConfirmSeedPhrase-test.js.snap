// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupConfirmSeedPhrase presentation renders correctly 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "#1c2227",
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#1c2227",
        "flex": 1,
        "paddingBottom": 26.5,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <Image
          resizeMode="contain"
          screenNumber={6}
          source={
            Object {
              "testUri": "../../../img/step-6.png",
            }
          }
          style={
            Object {
              "backgroundColor": "transparent",
              "height": 60,
              "marginBottom": 5,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 20,
              }
            }
          >
            Demonstrate that you wrote the phrase down by tapping the words below in order.
             
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                zero
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                one
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                two
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                three
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                four
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                five
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                six
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                seven
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                eight
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                nine
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                ten
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 6,
                  "flex": 1,
                  "height": 133.5,
                  "justifyContent": "center",
                  "marginBottom": 7.5,
                  "marginTop": 7.5,
                  "width": 225,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 20,
                    "textAlign": "center",
                  }
                }
              >
                eleven
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "space-between",
            "marginTop": 26.5,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityTraits="button"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "#4e957a",
              "borderRadius": 3,
              "height": 93.5,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                null,
                Object {
                  "color": "#fff",
                  "fontFamily": "TitilliumWeb-Regular",
                  "fontSize": 16,
                },
                null,
              ]
            }
          >
            Back (resets phrase)
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "justifyContent": "space-between",
            "marginTop": 26.5,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityTraits="button"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "#696969",
              "borderRadius": 3,
              "height": 93.5,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#dcdcdc",
                },
                Object {
                  "color": "#fff",
                  "fontFamily": "TitilliumWeb-Regular",
                  "fontSize": 16,
                },
                undefined,
              ]
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
